### PR TITLE
support name logic everywhere related to league

### PR DESCRIPTION
**Pull request by smallst**
we already use new name logic to leaderboards,
this pr make it also work in :
- MyMatchesTab.
- ladderPlayModal.
- playLevelView 
- spectateView

### CODE REVIEW

File: app/lib/LevelLoader.coffee [hunk 1]
```diff
@@ -261,6 +261,9 @@ module.exports = class LevelLoader extends CocoClass
       session.patch = session.save = session.put = -> console.error "Not saving session, since we didn't create it."
     else if codeLanguage = utils.getQueryVariable 'codeLanguage'
       session.set 'codeLanguage', codeLanguage
+    if me.id is session.get('creator')
+      # do not use .set since we won't update fullName to server
+      session.fullName = _.filter([me.get('firstName'), me.get('lastName')]).join(' ')
```
Comment (mrfinch) [inline app/lib/LevelLoader.coffee:266]: can't we use user.broadName ?

Comment (smallst) [reply app/lib/LevelLoader.coffee:266]: 🤔 actually we do part of `broadName` logic when displaying this fullName. And yes we can turn to broadName I believe.

Comment (mrfinch) [reply app/lib/LevelLoader.coffee:266]: It does things on top of your logic so it might make more consistent but anyway you have merged now 😅

Comment (smallst) [reply app/lib/LevelLoader.coffee:266]: ya 😂  i can update later if we find broadName is better. besides this pr there is another place we use simple `first last . join`

**Pull request merged**